### DISCUSSION TITLE
Update `esbuild` in `app-vite` to support TS 4.9

### DISCUSSION THREAD

**Discussion by mv-go** [quasarframework/quasar#15123] [Ideas / Proposals] [answered]
I tried using the new `satisfies` operator (came with TS 4.9) and `esbuild`, not surprisingly, failed to compile.

Should we update `esbuild` in `@quasar/app-vite` to, at least, [0.15.3](https://github.com/evanw/esbuild/releases/tag/v0.15.13)?

Currently it's `0.14.51` in app-vite while the latest is `0.16.7`.

**Comment by yusufkandemir** [comment, marked as answer]
`@quasar/app-vite` v1 uses `vite` v2, which uses `esbuild` v0.14:
https://github.com/vitejs/vite/blob/3a5543db215cedbcadcfa2e14a1c48b14acdd3c5/packages/vite/package.json#L46

`@quasar/app-vite` uses `esbuild` to build files that are separate from the main app, such as the BEX background script, the Electron main script, etc. So, any inconsistency between Quasar CLI building these with `esbuild` v0.15 and Vite building the app code with `esbuild` v0.14 could cause problems in a project out there, even if it seems to work at first. So, we have to wait until Quasar CLI with Vite has Vite 4 support, see: #14077

If you feel lucky or adventurous, you can make your package manager enforce the version you want, while waiting for Vite 4 support:

[Yarn - `resolutions`](https://classic.yarnpkg.com/lang/en/docs/selective-version-resolutions/#toc-how-to-use-it)
```json
"resolutions": {
  "esbuild": "0.15.3"
}
```

[NPM 8.3+ - `overrides`](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#overrides)
```json
"overrides": {
  "esbuild": "0.15.3"
}
```

**Reply by mv-go** [reply]
Much love to you, sir, for the detailed explanation! I will mark your reply as an `answer` and will patiently wait on Vite 4 support. Thanks again!
